cypress/support/step_definitions/dictionary/addBulkConceptsToDictionary/addBulkConceptsToDictionary.ts

```diff
@@ -0,0 +1,130 @@
+/// <reference types="cypress" />
+/// <reference types="../../../" />
+import { 
+  After, 
+  Before, 
+  Given, 
+  Then, 
+  When } from "cypress-cucumber-preprocessor/steps";
+import { getDictionaryId, getUser, getConceptId } from '../../../utils';
+
+Given("the user is on the view dictionary concepts page", () => {
+  cy.visit(`/users/${getUser()}/collections/${getDictionaryId()}/concepts/`);
+  cy.url().should(
+    "contain",
+    `/users/${getUser()}/collections/${getDictionaryId()}/concepts/`
+  );
+});
+
+Given('the user is on the "Add concepts in bulk from CIEL" page', () => {
+cy.visit(
+`/users/${getUser()}/collections/${getDictionaryId()}/add/?fromSource=CIEL/`
+);
+cy.url().should(
+"contain",
+`/users/${getUser()}/collections/${getDictionaryId()}/add/?fromSource=CIEL/`
+  );
+});
+
+Given('CIEL concept "1000" is already in the dictionary', () => {
+  cy.visit('/actions/');
+  cy.url().should(
+    "contain", '/actions/'
+  );
+});
+
+When('the user clicks the "Add concepts" floating action button', () => {
+  cy.findByTitle("Add concepts").click();
+});
+
+When('the user clicks the "Add concepts" MuiButton-label button', () => {
+  cy.findByText("Add concepts").click();
+});
+
+When('the user selects "Add bulk concepts"', () => {
+  cy.visit(
+    `/users/${getUser()}/collections/${getDictionaryId()}/add/?fromSource=CIEL/`
+  );
+});
+
+When(
+  /the user enters concept Id "(1000|1001|1002)"/, (conceptId) => {
+    cy.get("textarea").type("conceptId")
+  }
+); 
+
+When('the user re-enters concept Id "1000"', (conceptId) => {
+  cy.get("textarea").type("conceptId")
+});
+
+When('the user navigates to the "Progress notifications" page', () => {
+   cy.url().should("actions/");
+});
+
+Then('the user selects "Import existing concept"', () => {
+   cy.findByText("Import existing concept").click();
+});
+
+Then('the user should be on the "Add concepts in bulk from CIEL" page', () => {
+   cy.url().should("contain", `/users/${getUser()}/collections/${getDictionaryId()}/add/?fromSource=CIEL/`);
+});
+
+Then(
+  /the "(1000|1001|1002)" concepts should be added to the dictionary/, conceptId => {
+    cy.getConcept(`/users/${getUser()}/collections/${getDictionaryId()}/concepts/${conceptId}`).should("exist");
+    cy.get("textarea").type("conceptId")
```

Comment:
We don't need this line here. There's no text area. More generally `When()` steps describe an action the user does and `Then()` steps describe an assertion.

Reply:
Concerning this error `CypressError: `cy.type()` can only accept a string or number. You passed in: `undefined`` I thought I was to use a string in `.type("conceptId")` Kindly what is your take?

Reply:
My point was that we shouldn't be calling `cy.type()` here at all.